packages/block-editor/src/components/block-editing-mode/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+export function useBlockEditingMode( clientId, mode ) {
```

Comment:
There's no real benefit to using `useBlockEditingMode` instead of `getBlockEditingMode`. But yes the benefit of using `useBlockEditingMode` over `setBlockEditingMode` is that it handles setting and clearing the mode in the effect. Mostly this hook exists so that the API is as declarative for block authors as possible.